packages/sitecore-jss-angular/src/components/placeholder.component.ts

```diff
@@ -265,17 +347,30 @@ export class PlaceholderComponent implements OnInit, OnChanges, DoCheck, OnDestr
       this.isLoading = false;
     } else {
       const factories = await this.componentFactory.getComponents(placeholder);
-
       try {
         const nonGuarded = await this.guardResolver(factories);
         const withData = await this.dataResolver(nonGuarded);
+        // not using index to ensure code blocks are rendered at correct positions
```

Comment:
What happens in case we continue to use "index"? The index shouldn't affect and give you the wrong order. Also, I see that you made "index" optional in some functions, while we don't provide index anymore
```
index: The 0-based index at which to insert the new view into this container. If not specified, append the new view as the last entry.
```

Reply:
> What happens in case we continue to use "index"?

All code blocks will be rendered after all the components are added, instead of wrapping them